public/app/plugins/datasource/graphite/datasource.ts

```diff
@@ -255,6 +258,9 @@ export class GraphiteDatasource
       if (options.panelId) {
         httpOptions.headers['X-Panel-Id'] = options.panelId;
       }
+      if (options.panelPluginId) {
```

Comment:
Always wondered if we could automatically forward these contextual values from the query runner to the backend_srv somehow so each datasource doesn't have to care about this. Discussed historically as well, but without any good solution. Isn't there a `requestId` set for each query that query runner forwards to datasource  that can be connected to these contextual values as as example and allow the backend_src to lookup by `requestId`?

Reply:
Sorry didn't catch the `For backend data source the header is automatically added via DataSourceWithBackend.query method` until now. So I guess no longer that big of a problem so please disregard :smile:

Reply:
Yeah, it's trickier to do for front-end data sources as they usually use backend srv's `fetch` method which is a generalized way to get any arbitrary grafana endpoint results. I would rather avoid extending `BackendSrvRequest` interface with another property to identify source panel plugin id automatically when the data source can provide the headers already. It's suboptimal for front-end data sources, I know.

Reply:
Thought of direct/browser connected datasources and not using the datasource proxy. My bad. So for those using proxy they would need to manually forward this header